src/components/vm/VmInitializer.tsx

```diff
@@ -96,21 +98,21 @@ export default function VmInitializer() {
     });
   }, [near]);
 
-  const requestSignInWithWallet = useCallback(
-    (event: any) => {
-      event?.preventDefault();
-      walletModal?.show();
-      return false;
-    },
-    [walletModal],
-  );
+  const requestSignInWithWallet = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    walletModal?.show();
+    return false;
+  }, [signInRedirect, walletModal]);
 
-  const requestSignIn = useCallback(
-    (queryParam?: string) => {
-      router.push(`/signin${queryParam}`);
-    },
-    [router],
-  );
+  const requestSignIn = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    router.push('/signin');
+  }, [router, signInRedirect]);
+
+  const requestCreateAccount = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    router.push('/signup');
+  }, [router, signInRedirect]);
```

Comment:
But this two functions are the same but with different names why do we need both?